test/e2e/tests/settings/address-book.spec.js

```diff
@@ -70,6 +70,61 @@ describe('Address Book', function () {
       },
     );
   });
+
+  it('Adds a new contact to the address book', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.fullTitle(),
+      },
+      async ({ driver }) => {
+        await unlockWallet(driver);
+        await openMenuSafe(driver);
+
+        await driver.clickElement({ text: 'Settings', tag: 'div' });
+        await driver.clickElement({ text: 'Contacts', tag: 'div' });
+
+        await driver.clickElement('.address-book__link');
+
+        const inputUsername = await driver.findElement('#nickname');
+        await inputUsername.fill('#nickname', 'Test User');
+
+        const inputAddress = await driver.findElement(
+          'input[placeholder="Enter public address (0x) or domain name"]',
+        );
+        await inputAddress.fill(
+          '[data-testid="ens-input"]',
+          '0x56A355d3427bC2B1E22c78197AF091230919Cc2A',
+        );
+
+        await driver.clickElement('[data-testid="page-container-footer-next"]');
+
+        const recipientUsername = await driver.waitForSelector({
+          text: 'Test User',
+          css: '.address-list-item__label',
+        });
+
+        assert.equal(
+          await recipientUsername.getText(),
+          'Test User',
+          'Username is not added correctly',
+        );
+
+        const recipientAddress = await driver.waitForSelector({
+          css: '[data-testid="address-list-item-address"]',
+          text: '0x56A35...9Cc2A',
+        });
+
+        assert.equal(
+          await recipientAddress.getText(),
```

Comment:
Since we have added the `waitForSelector`

- we can remove the following code snippet mentioned below and 
- the variable declaration as well (const recipientAddress =)

```
assert.equal(
          await recipientAddress.getText(),
          shortenAddress('0x56A355d3427bC2B1E22c78197AF091230919Cc2A'),
          'Recipient address is not added correctly',
        );
```